torch/csrc/Module.cpp

```diff
@@ -175,6 +176,35 @@ static PyObject * THPModule_crashIfATenASAN(PyObject *module, PyObject *arg) {
   return THPUtils_packInt32(at::_crash_if_asan(THPUtils_unpackInt(arg)));
 }
 
+static PyObject * THPModule_vitals_enabled(PyObject *module, PyObject *args) {
```

Comment:
These would be much cleaner with PyBind11. (`py_module.def`)

Reply:
Do you have an example? I adopted what I saw used other places in this file.

Reply:
Something like https://github.com/pytorch/pytorch/blob/master/torch/csrc/Module.cpp#L934

In your case I think it would be as simple as:
```
py_module.def("vitals_enabled", &at::vitals::torchVitalEnabled);
py_module.def("set_vital", &at::vitals::VitalsAPI.setVital);
```

PyBind11 is great and will figure out all the arguments with template matching.

Reply:
Ah, so basically a different way to register instead of using?
static PyMethodDef TorchMethods[] = { ... }

I'll take a look!

Reply:
"Cannot create a non-constant pointer to member function" -- had to us a lambda.

But wow - this is like magic. it literally did everything for me.